### ISSUE TITLE
🐛 BUG: --- in body of markdown causes a crash

### ISSUE THREAD

**Issue by stenehall** [snowpackjs/astro#720]
### What package manager are you using?

yarn

### What operating system are you using?

Mac

### Describe the Bug

Having `---` in the body of a markdown post will crash the build / dev server

```
---
title: 'More astro'
---

There's a simple secret to building a faster website — _just ship less_.

With this in the body ---

Unfortunately, modern web development has been trending in the opposite direction—towards _more._ More JavaScript, more features, more moving parts, and ultimately more complexity needed to keep it all running smoothly.
```

### Steps to Reproduce

1. `npm init astro` using template <NAME>
2. add `---` to the body of a markdown post (see bug description)
3. `yarn start` or `yarn build`
5. Error! Getting a "Javascript heap out of memory" error when trying to build

**Comment by jasikpark** [comment]
I wonder if this is a bug with the astro parser or with remark?

**Comment by FredKSchott** [comment]
This is probably on our end. I believe `---` is allowed in the markdown body to denote a `<hr/>` element.

**Comment by brycewray** [comment]
While this has been fixed in Astro 0.18.5, the SmartyPants functionality isn't able to recognize `---` as something it should turn into an em dash, while it turns `--` into an em dash rather than an en dash. Probably needs a way to set the SmartyPants options for old-school dashes: i.e., ```{ dashes: 'oldschool' }``` (https://github.com/silvenon/remark-smartypants).
